### ISSUE TITLE
Installing itch game with uri secret

### ISSUE THREAD

**Issue by local-minimum** [itchio/itch#1080]
If you have access to a game with downloads and copy paste the full URI including the `?secret=...`, the itch app gives you an error page.

The game in question has publicly announced the secret on Ludum Dare and should probably rather be published, but I figure it would still be very nice if you were sent a secret URI for an itch game to be able to install it via the app.

I don't want to directly post the secret here, so I link to the Ludum-page where I found the game:

[The Ludum Page that was/is secret-linking.](http://ludumdare.com/compo/ludum-dare-37/?action=preview&uid=45525)

**Comment by fasterthanlime** [comment]
I started taking care of it https://github.com/itchio/itch/commit/4cb763b1d5d74cbf80e72583be24ea5ff242b044

**Comment by leafo** [comment]
keep in mind the secret and the password are two separate things:

* `password` for restricted games (value provided by user)
* `secret` is for draft games (value autogenerated by itch.io)

draft games are kinda broken right now: If you don't own the game but have the secret URL, you're not allowed to download or buy the game. This is because the secret URL's intent was just for sharing a preview of what the page would look like.

I think if you want to share a full game to a restricted audience then I recommend using restricted mode along with a password (or by generating download keys for them)

**Comment by fasterthanlime** [comment]
@leafo yeah, the linked commit doesn't really do my wip justice - I did have password/secret confused originally, but handle both now (and so does @granpc's PR)

**Comment by fasterthanlime** [comment]
@leafo Btw, about restricted games: the app will still err with invalid games, since it can't use download keys when querying `/game/:game_id` - how do we solve that? Should the API check if the API user owns the game? (Or.. does it already do that? In which case I'm confused about the support ticket)

**Comment by triplefox** [comment]
Ran into this issue today: The expectation from the game's author was that the experience for a game in draft mode accessed by itch app would be like distributing builds from e.g. TestFlight.

**Comment by fasterthanlime** [comment]
Just to clarify: currently, both `password`s (restricted games) and `secret`s (draft games) are unsupported in the itch app.

I've been thinking of ways to solve this server-side, though, the client-side fixes I had in mind are.. not great.

**Comment by Noobidoo** [comment]
@fasterthanlime Any progress on this?

My thought on how this could work:
1)The api should get the associated shared key sent to it and you can there decide if there is enough permission or not.
2) the game gets associated with the user when he adds it to his collection. Here you get problems with revoking keys to games, and need to explicitly remove the game from each collection.
3) When it comes to 'download key' allow several user to claim the same key and just track with user that has claimed it.
